solutions/gold/usaco-1018.mdx

```diff
@@ -111,3 +121,100 @@ print(ans, file=open("help.out", "w"))
 
 </PySection>
 </LanguageSection>
+</Spoiler>
+
+<Spoiler title="Solution 2">
+
+<Warning>
+
+This solution makes use of [indexed sets](https://usaco.guide/gold/PURS?lang=cpp#finding-the-k-th-element).
+
+</Warning>
+
+## Explanation
+
+We process the segments one-by-one. Let the complexity sum of the the first $k$ segments be $c_k$. We'll use $c_k$ to calcualte $c_{k+1}$ in a process similar to dynamic programming. Our answer would then be $c_N$.
+
+From $c_k$ to $c_{k+1}$, our new set of subsets is,
+
+$$\text{\{the previous set of subsets\} } \cup \text{ \{the previous set of subsets with }segment_{k+1}\text{ added to each of them\}}$$
+
+The sum of the complexities of the previous set of subsets is still $c_k$. However, adding $segment_{k+1}$ to each of them may merge lots of previously disjoint segments (if $segment_{k+1}$ happened to intersect all of them).
+
+To control this, **we first sort the segments by their left endpoints**. This means that each new segment can only either merge itself into a previous segment (without merging previously disjoint segments) or form a new disjoint segment.
+
+We also know that our new segment forms a new disjoint segment if and only if it merges with previous segments that doesn't intersect with it. Therefore, the transition formula is,
+
+$c_{k+1} = c_k + 2^{\text{number of previous segments that don't intsersect } segment_k}$
```

Comment:
eek it's kinda hard to read a \text this long, maybe you could say something like, if x is the number of prev segments ..., and then just write 2^x  or smtg